### PR TITLE
Update docs related to generate() method in transformers/pipelines

### LINKED ISSUE #28224: [DOCS] - Wrong link to model#generative-models in Pipeline docs
### System Info

Latest docs on https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.TextGenerationPipeline

### Who can help?

@stevhliu and @MKhalusova

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Inside the documentation for [pipelines#transformers.TextGenerationPipeline.__call__](https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.TextGenerationPipeline.__call__) the docs read:

> generate_kwargs — Additional keyword arguments to pass along to the generate method of the model (see the generate method corresponding to your framework [here](https://huggingface.co/docs/transformers/en/main_classes/model#generative-models)).

The header part of the link (#generative-models) does not exist in the [main model page](https://huggingface.co/docs/transformers/en/main_classes/model) though - I'm not sure any of the headers there are appropriate actually. 

The problem is repeated for two other tasks at least: Conversational, text2textgeneration.

For the text-generation case, the corresponding docs link is [pipelines/text_generation.py#L198](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/text_generation.py#L198)

### Expected behavior

The link sends me to documentations about common model-generation parameters - Maybe [this](https://huggingface.co/docs/transformers/generation_strategies)?

### PR DESCRIPTION
**Pull request by coolyashas**
# What does this PR do?

Fixes #28224

In the issue, it is suggested to change the link to [this](https://huggingface.co/docs/transformers/generation_strategies), but I find [this](https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/text_generation) to be a better fit.

@stevhliu and @MKhalusova

### CODE REVIEW

**Review by stevhliu — APPROVED** [review]
Thanks for the fixing the link!

**Comment by coolyashas** [comment]
I believe all necessary changes have been made @stevhliu

**Comment by stevhliu** [comment]
Hey, can you also apply the changes to the `text2text_generation.py` and `text_generation.py` files please?

**Comment by coolyashas** [comment]
> Hey, can you also apply the changes to the `text2text_generation.py` and `text_generation.py` files please?

image

Hey, the changes have already been made in the files you mentioned in this PR itself [here](https://github.com/huggingface/transformers/pull/28231/files). Are there any other changes you wanted implemented?

File: src/transformers/pipelines/text2text_generation.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def __call__(self, *args, **kwargs):
                 max_length instead of throwing an error down the line.
             generate_kwargs:
                 Additional keyword arguments to pass along to the generate method of the model (see the generate method
-                corresponding to your framework [here](./model#generative-models)).
+                corresponding to your framework [here](https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/text_generation)).
```
Comment (stevhliu) [inline src/transformers/pipelines/text2text_generation.py:157]:
```suggestion
                corresponding to your framework [here](./main_classes/text_generation)).
```

File: src/transformers/pipelines/text2text_generation.py [hunk 2]
```diff
@@ -257,7 +257,7 @@ def __call__(self, *args, **kwargs):
                 Whether or not to clean up the potential extra spaces in the text output.
             generate_kwargs:
                 Additional keyword arguments to pass along to the generate method of the model (see the generate method
-                corresponding to your framework [here](./model#generative-models)).
+                corresponding to your framework [here](https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/text_generation)).
```
Comment (stevhliu) [inline src/transformers/pipelines/text2text_generation.py:260]:
```suggestion
                corresponding to your framework [here](./main_classes/text_generation)).
```

File: src/transformers/pipelines/text2text_generation.py [hunk 3]
```diff
@@ -359,7 +359,7 @@ def __call__(self, *args, **kwargs):
                 for single pair translation models
             generate_kwargs:
                 Additional keyword arguments to pass along to the generate method of the model (see the generate method
-                corresponding to your framework [here](./model#generative-models)).
+                corresponding to your framework [here](https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/text_generation)).
```
Comment (stevhliu) [inline src/transformers/pipelines/text2text_generation.py:362]:
```suggestion
                corresponding to your framework [here](./main_classes/text_generation)).
```

File: src/transformers/pipelines/text_generation.py [hunk 4]
```diff
@@ -195,7 +195,7 @@ def __call__(self, text_inputs, **kwargs):
 
             generate_kwargs:
                 Additional keyword arguments to pass along to the generate method of the model (see the generate method
-                corresponding to your framework [here](./model#generative-models)).
+                corresponding to your framework [here](https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/text_generation)).
```
Comment (stevhliu) [inline src/transformers/pipelines/text_generation.py:198]:
```suggestion
                corresponding to your framework [here](./main_classes/text_generation)).
```

**Review by stevhliu — APPROVED** [review]
Highlighted a couple more places where we should also use the local link :)

**Comment by coolyashas** [comment]
@stevhliu I deleted this repo mistakenly. Could you pls reopen it in order to finish the PR?

**Comment by amyeroberts** [comment]
@coolyashas Unfortunately we can't reopen this PR, as the repo through which this PR was submitted has been deleted. If you reopen another PR with the same changes I'll be happy to do a quick review.

**Comment by coolyashas** [comment]
> @coolyashas Unfortunately we can't reopen this PR, as the repo through which this PR was submitted has been deleted. If you reopen another PR with the same changes I'll be happy to do a quick review.

Sure will do, thanks for the reply!

**Pull request closed without merging**